proxy/testing/proxy.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2021 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package testing
+
+type MockProxier struct {
```

Comment:
ideally gomock for new mocks

Reply:
These don't need to be mocked. They are far to simple and introduces an unneeded dependency.